patching/config/src/getPatchInfo.ts

```diff
@@ -0,0 +1,30 @@
+import { PnpmError } from '@pnpm/error'
+import { type ExtendedPatchInfo, type PatchGroupRecord } from '@pnpm/patching.types'
+import { satisfies } from 'semver'
+
+export function getPatchInfo (
+  patchFileGroups: PatchGroupRecord | undefined,
+  pkgName: string,
+  pkgVersion: string
+): ExtendedPatchInfo | undefined {
+  if (!patchFileGroups?.[pkgName]) return undefined
+
+  const exactVersion = patchFileGroups[pkgName].exact[pkgVersion]
+  if (exactVersion) return exactVersion
+
+  const versionRanges = Object
+    .keys(patchFileGroups[pkgName].range)
```

Comment:
why is range a map? it looks like an array would work better.

Reply:
`PatchGroup.range` maps `pkg@version-range` to patch file paths. I temporary converted it to an array merely to detect conflicts. "Conflicts" here means that there are more than 1 version range that may apply to one package as explained in [this comment](https://github.com/pnpm/pnpm/pull/9236#discussion_r1985210936).

Reply:
[It does make the code a bit more verbose](https://github.com/pnpm/pnpm/pull/9236/commits/b3b2bb3e37c75101fe639ec64431db699936bf8a)